### PR TITLE
solana: init at 1.5.19

### PR DESCRIPTION
**Pull request by ruuda**
This adds [Solana](https://solana.com/), a performance-oriented blockchain. The source is the Solana monorepo, which includes the development tools and tools needed to interact with the network.

Version 1.5.19 is not the latest release at the time of writing, but it is the latest one intended for use with the mainnet.

###### Motivation for this change

* Having the Solana devtools available in Nixpkgs makes it easy to manage development environments with Nix.
* Solana users who also use Nix don’t need to download a binary / manually build from source any more.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/solana/default.nix [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    rocksdb
+  ];
+
+  LLVM_CONFIG_PATH = "${llvm}/bin/llvm-config";
+  LIBCLANG_PATH = "${llvmPackages.libclang}/lib";
+
+  # Used by build.rs in the rocksdb-sys crate. If we don't set these, it would
+  # try to build RocksDB from source.
+  ROCKSDB_INCLUDE_DIR = "${rocksdb}/include";
+  ROCKSDB_LIB_DIR = "${rocksdb}/lib";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/solana/default.nix:49]: Can we export those in preBuild?

Comment (ruuda) [reply pkgs/applications/blockchains/solana/default.nix]: Done.

Note that this PR is a work in progress, the Rust binaries build now, but the `cargo-bpf` tools refer to the BPF SDK which is not yet packaged. (Upstream there’s a script that downloads prebuilt binaries from GitHub, the binaries include a patched Rust + LLVM toolchain.)

**Comment by rollschild** [comment]
Hi, has there been any progress on this recently?

**Comment by NickCao** [comment]
superseded by later PRs.

**Pull request closed without merging**